### PR TITLE
Add utility functions for commit review status pipline

### PR DESCRIPTION
**Pull request by lock14**
* Added function for retrieving the first approving PR from a slice of PRs
* Added a function to construct a commit's github url
* Added a convenience function to construct a GitHub GraphQL client given an access token

### CODE REVIEW

File: pkg/review/commit_review_status.go [hunk 1]
```diff
@@ -22,8 +22,13 @@ import (
 	"fmt"
 
 	"github.com/shurcooL/githubv4"
+	"golang.org/x/oauth2"
 )
 
+// githubPrApproved is the value GitHub set the `reviewDecision` field to
+// when a Pull Request has been approved by a reviewer.
+const githubPrApproved = "APPROVED"
```
Comment (drevell) [inline pkg/review/commit_review_status.go:30]: The conventional Go capitalization would be "githubPRApproved", per https://google.github.io/styleguide/go/decisions#initialisms

File: pkg/review/commit_review_status.go [hunk 2]
```diff
@@ -61,6 +66,16 @@ WHERE
   AND issues.closed_at >= TIMESTAMP('%s')
 `
 
+// Commit maps the columns from the driving BigQuery query
+// to a usable structure.
+type Commit struct {
+	Author       string `bigquery:"author"`
+	Organization string `bigquery:"organization"`
+	Repository   string `bigquery:"repository"`
+	Sha          string `bigquery:"commit_sha"`
```
Comment (drevell) [inline pkg/review/commit_review_status.go:75]: Ditto here, Sha -> SHA

File: pkg/review/commit_review_status.go [hunk 3]
```diff
@@ -61,6 +66,16 @@ WHERE
...
+	Timestamp    string `bigquery:"commit_timestamp"`
```
Comment (drevell) [inline pkg/review/commit_review_status.go:76]: Perhaps consider adding a comment here that explains the format of the timestamp (millis, micros, RFC1234, etc).

File: pkg/review/commit_review_status_test.go [hunk 4]
```diff
@@ -533,6 +533,108 @@ WHERE
...
+		},
+	}
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			got := getApprovingPullRequest(tc.pullRequests)
+			if diff := cmp.Diff(got, tc.want); diff != "" {
+				t.Errorf("getCommitHTMLURL unexpected result (-got,+want):\n%s", diff)
```
Comment (drevell) [inline pkg/review/commit_review_status_test.go:602]: This might be a copy-paste oversight, getCommitHTMLURL -> getApprovingPullRequest

**Review by drevell — CHANGES_REQUESTED** [review]

File: pkg/review/commit_review_status.go (hunk 1 above)
Comment (lock14) [reply pkg/review/commit_review_status.go]: done.

File: pkg/review/commit_review_status.go (hunk 2 above)
Comment (lock14) [reply pkg/review/commit_review_status.go]: done.

File: pkg/review/commit_review_status.go (hunk 3 above)
Comment (lock14) [reply pkg/review/commit_review_status.go]: done.

File: pkg/review/commit_review_status_test.go (hunk 4 above)
Comment (lock14) [reply pkg/review/commit_review_status_test.go:602]: fixed.

**Review by drevell — APPROVED** [review]

**Pull request merged**